doc/plutus-core-spec/introduction.tex

```diff
@@ -18,3 +18,51 @@ \section{Introduction}
 untyped Plutus Core is used as a compilation target, and we will eventually
 provide a specification of the type system and semantics of Typed Plutus Core
 here as well, together with its relationship to Untyped Plutus Core.
+
+\subsection{Terminology and version numbers}
+\label{sec:version-numbers}
+Once a Plutus Core script has been deployed to the blockchain, it may be
+re-evaluated at any point in the future when the history of the chain is
+verified.  Plutus Core may evolve as time progresses, and because of this we
+have to be quite careful with versioning.  A detailed discussion of this issue
+can be found in~\cite{CIP-35}, but we reproduce two definitions here for
+convenience.
+\begin{itemize}
+\item The language itself may evolve: for example some fundamental new construct
+  may be added.  A \textit{Plutus Core language version} describes a version of
+  the basic language with a particular set of features. A language version
+  consists of three non-negative integers separated by decimal points, for
+  example \texttt{1.4.2}.  Language versions are ordered lexicographically.
+\item The built-in types and functions deployed on the chain along with a
+  particular language version can change with time, and we refer to Plutus Core
+  along with a particular set of types and functions as a \textit{Plutus Core
+    ledger language version}.  These are referred to using version names like
+  ``Plutus V1,'', ``Plutus V2'', and so on.
+  \end{itemize}
+ \noindent These two kinds of version are independent: all Plutus Core language
+ versions must support all ledger language versions.
```

Comment:
I think this isn't exactly true.  For example, the V1 cost parameters (as stored in the protocol parameters) don't contain a cost for `cekCaseCost`, so you can't use V1 with PLC 1.1.0.

Reply:
^ I now think that the previous comment is wrong.  The `defaultCostModel` for 1.1.0 contains entries for `cekCaseCost` and `cekConstrCost`, and if we use [applyCostModelParams](https://github.com/input-output-hk/plutus/blob/b321575d9266b3358b9e728d064fc0bee4f355d7/plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/CostModelInterface.hs#L273) to update that with an old set of V1 parameters `P` that don't contain those, there won't be an error: the entries for the original constructors like `cekApplyCost` will be overwritten with the corresponding entries from `P`, but the entries for `cekCaseCost` and `cekConstrCost` will be left untouched: see [this test](https://github.com/input-output-hk/plutus/blob/b321575d9266b3358b9e728d064fc0bee4f355d7/plutus-core/plutus-core/test/CostModelInterface/Spec.hs#L128) for example.

Reply:
I am also not sure anymore. Is it going to be possible to run V1 together with plc-1.1.0? @michaelpj

Reply:
I'll revise that when I work out what the truth is.